### PR TITLE
Plugins/Themes: replace upload eligibility error message with upsell

### LINKED ISSUE #33063: Plugins: Upload plugins page says there's an error if you don't have a Business plan
Screen Shot 2019-05-15 at 12 21 24 PM

If you try to upload a plugin when you don't yet have a Business plan, you see this page.

* "The errors above must be resolved before proceeding." - This message is generic and not very helpful. If this is an error, it should display with the red indicators/icons around it. 
* That said, I don't think we need this message at this stage in the process. If the goal is to drive people to the Business plan from this page, we shouldn't distract them with extraneous messaging. 
* This page needs a clearer call to action. The upgrade notice uses small type, and the wording doesn't mention plugins at all -- it distracts with "also get...." a bunch of features that distract from the ultimate goal, which on this page is to upload plugins.

Found during this walkthrough by @ianstewart https://cloudup.com/cTAV7Njdy1z

### LINKED ISSUE #33069: Themes: Uploads page shows bogus error
(Similar to #33063 for Plugins.) On the Upload Themes page, when you don't yet have a Business plan, the upgrade banner is shown, like this:

Screen Shot 2019-05-15 at 2 55 28 PM

* The copy mentions "Also get unlimited themes", when we're already on the Themes page, which is redundant.
* The overall design makes this banner look very small. It doesn't look like a clear call to action to upgrade; more like a suggestion.
* The "error" that doesn't look like an error is discouraging and doesn't provide much help. A clearer call to action should be provided. If the goal is to drive people to the Business plan from this page, we shouldn't distract them with extraneous messaging.

Found during this walkthrough by @ianstewart https://cloudup.com/iLSE_iIViNf

### PR DESCRIPTION
**Pull request by donpark**
#### Changes proposed in this Pull Request

When site lacks business plan, both plugin and theme upload pages currently display confusing error messages.

This patch adds upsell support to `EligibilityWarnings` class and specifies:

- `PluginsUpsellComponent` as `upsell` component in plugins upload page.
- `ThemesUpsellComponent` as `upsell` component in themes upload page.
- Tweaks upsell components' CSS styles to fit overall page style.

#### Design

Both `PluginsUpsellComponent` and `ThemesUpsellComponent` appear to have been designed for different use-case than plugins/themes upload pages so their styles had to be tweaked to be less _loud_.

#### Testing instructions

1. Test with a Simple site without Business plan.
2. Navigate to http://calypso.localhost:3000/plugins/upload/{site_slug}

*Before:*

*After:*
screenshot_863

3. Navigate to http://calypso.localhost:3000/themes/upload/{site_slug}

*After:*
screenshot_865

*

Fixes #33063
Fixes #33069

### CODE REVIEW

**Comment by donpark** [comment]
PR recreated with different branch name to avoid `add/some-feature` requiring test suites.

**Review by Aurorum — COMMENTED** [review]
> Upgrading to the Business plan unlocks access to more than 50,000 WordPress Plugins and **197** premium Themes

Where's this number from? I know it's not being added in this PR specifically, but the Theme Directory says there's 183 themes. Not a big issue tho 

Screenshot 2019-08-22 at 09 59 57

> Both PluginsUpsellComponent and ThemesUpsellComponent appear to have been designed for different use-case than plugins/themes upload pages so their styles had to be tweaked to be less loud.

I still personally feel like these are still a bit...aggressive. That may be the intention, though I would've imagined an illustration to prevent the screen being dominated by a big primary CTA. Also, I would've imagined the plan features icon being aligned to the left instead of being at the top. 

Screenshot 2019-08-22 at 09 52 12

But those are just my thoughts on how I would've imagined preventing the upsells from being a bit too loud. Definitely not something that needs to be implemented! :) 

One quick comment below too

File: client/blocks/eligibility-warnings/index.jsx [hunk 1]
```diff
@@ -124,34 +125,36 @@ export const EligibilityWarnings = ( {
...
-
-					<Button primary={ true } disabled={ ! isEligible } onClick={ onProceed }>
-						{ translate( 'Proceed' ) }
-					</Button>
-				</div>
-			</Card>
+			{ isEligible && (
+				<Card className="eligibility-warnings__confirm-box">
+					<div className="eligibility-warnings__confirm-text">
+						{ isEligible &&
```
[the remark this answers is not in the document]
Comment (donpark) [reply client/blocks/eligibility-warnings/index.jsx]: goog catch. updated.

**Comment by donpark** [comment]
> > Upgrading to the Business plan unlocks access to more than 50,000 WordPress Plugins and **197** premium Themes
> 
> Where's this number from? I know it's not being added in this PR specifically, but the Theme Directory says there's 183 themes. Not a big issue tho

I've no idea. I just found the two, seemingly no longer used anywhere, `UpsellBlahComponent` classes and restyled them for inline use.

> I still personally feel like these are still a bit...aggressive. That may be the intention, though I would've imagined an illustration to prevent the screen being dominated by a big primary CTA. Also, I would've imagined the plan features icon being aligned to the left instead of being at the top.
...
> But those are just my thoughts on how I would've imagined preventing the upsells from being a bit too loud. Definitely not something that needs to be implemented! :)

Hmm. I was going to just deploy what's there now then make changes based on feedback but requesting reviews from marketing and designers beforehand like sane folks do.

**Comment by victorespigares** [comment]
@donpark I can take a look to the copy on behalf of the Upsell team. What's the estimated deadline for this?

**Comment by donpark** [comment]
> @donpark I can take a look to the copy on behalf of the Upsell team. What's the estimated deadline for this?

No deadline AFAIK. This is just a part of FixTheFlows project.

> > Two upsell sections seems to have already been used in production before, although not inline nor within plugins/themes uploads pages
> 
> Can you point me to where else are these two upsell sections used?

Good question. Only places I can find are two page routes
conditionally defined in `client/my-sites/feature-upsell/index.js`
behind `config.isEnabled( 'upsell/nudge-a-palooza' )` config flag.

Routes are:
- `'/feature/plugins/:domain'`
- `'/feature/themes/:domain'`

**Comment by victorespigares** [comment]
Thanks, @donpark! I've been taking a deep look into the nudge-a-palooza project where this came from. 

Copy looks good for now, but it'd be great to have this launched as an AB test (well, two actually, one for plugins and one for themes), definitely not with the intention to find a winner, but more to measure and help us understand the impact, so we can establish a baseline. Then later we can keep iterating on the copy as needed. 

Let me know your thoughts. If needed, I can help with the logistics around it (marketingtesting p2 posts, etc).

**Comment by donpark** [comment]
> Copy looks good for now, but it'd be great to have this launched as an AB test (well, two actually, one for plugins and one for themes), definitely not with the intention to find a winner, but more to measure and help us understand the impact, so we can establish a baseline. Then later we can keep iterating on the copy as needed.

Sure. I'm OK with launching this as two AB tests.
 
> Let me know your thoughts. If needed, I can take care of the logistics around it (marketingtesting p2 posts, hypotheses, etc).

That'd make things go a lot smoother. Thx!

**Comment by donpark** [comment]
@victorespigares Sorry for the delay. I've added A/B test rigging to the PR but I'm stumped as to what else needs to be done. Am I missing some on the backend side?

Test names are `upsellUploadPlugin` and `upsellUploadTheme`.

**Comment by victorespigares** [comment]
@donpark no worries! 

Can we make the top banner in the variant pointing to the Checkout page with the Business plan on it? 

So both the main CTA button and the banner will point to the same place. 

I think it'd make more sense as this is the same behaviour as seen in the banner of the Plugins page (one step before Upload). And same for Themes. 

I've started looking at it to change it, but I'm not familiar with this part of Calypso and got lost pretty quickly. Let me know if is a quick change you can do, if not we can ship it as is. No need to change the banner link in the control though since it is how right now is working.

In the meantime, I'll start putting together the MarketingTesting P2 Post for these two tests so they're ready to go 👍

**Comment by donpark** [comment]
> Can we make the top banner in the variant pointing to the Checkout page with the Business plan on it?

We can but the primary button is presented as an action while the banner is presented as informational (equivalent of Learn More).

How about removing the banner link if `upsell` is present, leaving it as is for `control` case? Are there technical reasons we can't do that?

**Comment by donpark** [comment]
@victorespigares I've updated the patch so that the banner link is disabled if upsell is present on the page. This is what the banner looks like on both pages:

screenshot_936

screenshot_937

Text copy on the banner needs some changing but we can address that later. Let me know what you think.

**Comment by donpark** [comment]
@victorespigares this PR now implements the behavior you described: both the banner and primary action button goes to checkout.

**Comment by donpark** [comment]
@lcollette I've removed the "Needs Design Review" label because a) upsell content has already been used before, and b) we want to do A/B test before iterating on the upsell content.

**Comment by shaunandrews** [comment]
I think the label on the button is way too long. Perhaps it can be shortened to "Upgrade to Business".

I think overall, all the copy needs some review and updates — the button label stands out more than the rest though.

**Comment by lcollette** [comment]
@scalarbane2 is going to take a look at the design.

**Comment by donpark** [comment]
@victorespigares We'll have to defer the A/B test until design issues are sorted out.

**Comment by lcollette** [comment]
@scalarbane2 Made some callouts for this: https://www.figma.com/file/BUi1CPAr8658TS7pcc4wH4/WordPress.com-UX-Audit?node-id=299%3A0

**Comment by victorespigares** [comment]
@lcollete @scalarbane2 Thanks for the inputs and great UX audit in that Figma! 👏👏👏 Looking forward to the day where all those are blew away from our user experience 🙏  

Regarding this PR: What are the next steps from Design here? I'm guessing the callouts for this would have to be crystallized in a proposal first and then be built at some point, right?

As previously noted, I agree both copy and design need some serious work here. 

But the whole idea was to launch this as a test to collect data to tackle that work better. 

With data, we'd know if this approach is actually improving metrics and is worth iterating on (or quite the opposite, and is better to look for other ideas) and also how critical are these two pages for upgrades and sales (so we know how much resources to put in there).

Since the PR is ready, I'm all for launching this as-is, gather some data and take the next steps from there. 

Personally, I'd say to hold on on any design work until we gather some data from this to know what the best next steps are. But if you want to go ahead and start putting together a new design, in the meantime this test can be running and gathering data as well :)

Let me know your thoughts!

**Comment by lcollette** [comment]
> Regarding this PR: What are the next steps from Design here? I'm guessing the callouts for this would have to be crystallized in a proposal first and then be built at some point, right?

There are a couple approaches we could take here. At minimum, what the user can do here (install plugin or theme) needs to be clearly communicated before any kind of upsell. The user may have some expectations we need to acknowledge to avoid cognitive hits. 

>As previously noted, I agree both copy and design need some serious work here.

My suggestion is we pause on this, step back, and do the serious work that copy and design needs. We, as a company, have to stop shipping things that are just okay to see what sticks. These micro-moments can negatively affect our customers, alienating them from using our product. 

Replacing the plugins and themes install pages feels aggressive and hurried. It feels like an unexpected marketing email. Might there be another way to satisfy your goals? 

Can we break down your intent with each section/copy block? That would help us structure the page a bit more hierarchical. If we know users, generally, prefer scanning over reading paragraphs, how about we edit down the copy to be more digestable? This might help convert better.

>Personally, I'd say to hold on on any design work until we gather some data from this to know what the best next steps are. But if you want to go ahead and start putting together a new design, in the meantime this test can be running and gathering data as well :)

I strongly disagree here. I feel like holding ourselves to a higher quality of work, carefully considering the user experience, and respecting our customers is really important. I don't feel we need additional data to tell us what's wrong here.

**Comment by sixhours** [comment]
> My suggestion is we pause on this, step back, and do the serious work that copy and design needs. We, as a company, have to stop shipping things that are just okay to see what sticks. These micro-moments can negatively affect our customers, alienating them from using our product.

Agreed. If we're going to add an upsell here, let's do it in a way that doesn't confuse the surrounding UX. I'd also want to get Editorial to sign off on any copy, as I know we've had

**Comment by donpark** [comment]
I'm unassigning myself since Friday is my last day and it doesn't look like this will be deployed anytime soon. thx y'all. :wave:

**Comment by lcollette** [comment]
@donpark @victorespigares Give me 2-3 hours to work on the design. :)

**Comment by sirreal** [comment]
Is this worth keeping or can it be closed?

**Comment by donpark** [comment]
Hi Jon. I suggest closing but keeping or replacing the task in a TODO list somewhere since this is a loose-end that could have high-impact.

**Comment by sixhours** [comment]
New pods are looking at this as part of design selection and plugins on Atomic, so I'm closing this.

**Pull request closed without merging**